include/mata/noodlify.hh

```diff
@@ -40,6 +40,34 @@ namespace SegNfa
  */
 AutSequence noodlify(const SegNfa& aut, Symbol epsilon, bool include_empty = false);
 
+/**
```

Comment:
Actually, I don't understand what function does. Why is it equation? How do lhs and rhs relate and how are resulting noodles created w.r.t. to parameters?

Reply:
The same applies to python binding function. I would like better explanation of this function.

Reply:
This function is a wrapper for several operations used together. This way, the user does not have to call each operation separately from Python, passing the computed epsilon symbol around (somehow) and calling the next C++ operation with the return values of the previous operation.

libMata cannot work with equations, queries etc. Hence, we compute the noodles for the equation, but represent the equation in a way that libMata understands. The left side automata represent the left side of the equation and the right automaton represents the right side of the equation. To create noodles, we need a segment automaton representing the intersection. That can be achieved by computing a product of both sides. First, the left side has to be concatenated into a single automaton to compute the intersection on, though.

I can add this whole previous paragraph to a documentation for this function. Nevertheless, does this answer your question?

Reply:
Yes, that paragraph makes sense to me.

Reply:
Fixed. Closing the discussion for now.